app/src/main/java/com/jwisozk/flightreviews/ParamViewModel.kt

```diff
@@ -0,0 +1,125 @@
+package com.jwisozk.flightreviews
+
+import android.widget.Button
+import android.widget.EditText
+import androidx.annotation.NonNull
+import androidx.annotation.Nullable
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.jwisozk.flightreviews.util.SingleEventLiveData
+import kotlinx.coroutines.launch
+
+class ParamViewModel : ViewModel() {
+
+    companion object {
+        data class Input(val repository: Repository)
+    }
+
+    @Nullable
+    private var input: Input? = null
+
+    private val _parametersLiveData = MutableLiveData<Parameters>()
+    val parametersLiveData: LiveData<Parameters> = _parametersLiveData
+
+    private val _snackBarLiveData = SingleEventLiveData<String>()
+    val snackBarLiveData: LiveData<String> = _snackBarLiveData
+
+    private val _isLoadingLiveData = SingleEventLiveData<Boolean>()
+    val isLoadingLiveData: LiveData<Boolean> = _isLoadingLiveData
+
+    private val _isRefreshLiveData = SingleEventLiveData<Boolean>()
+    val isRefreshLiveData: LiveData<Boolean> = _isRefreshLiveData
+
+    private val _editTextLiveData = SingleEventLiveData<EditText>()
+    val editTextLiveData: LiveData<EditText> = _editTextLiveData
+
+    private val _submitButtonLiveData = SingleEventLiveData<Button>()
+    val submitButtonLiveData: LiveData<Button> = _submitButtonLiveData
+
+    fun setEditTextLiveData(editText: EditText) {
+        _editTextLiveData.value = editText
+    }
+
+    fun setSubmitButtonLiveData(button: Button) {
+        _submitButtonLiveData.value = button
+    }
```

Comment:
`ViewModel` cannot hold references to any `View`s. It's a huge anti-pattern and creates a memory leak. Instead `ViewModel` should just receive info that something changed via some methods (e.g. `onSubmitButtonClick`, `onRatingChange`, etc.) and updates data in the `LiveData`